source/upgrade.txt

```diff
@@ -81,61 +141,99 @@ This driver version introduces the following breaking changes:
   - ``BSONCXX_POLY_USE_SYSTEM_MNMLSTC``.
   - ``BSONCXX_POLY_USE_BOOST``.
 
+API Breaking Changes in v4.0
+````````````````````````````
+
+The v4.0 driver introduces the following breaking changes to the API:
+
+- Drops support for configuring ``bsoncxx::stdx::string_view`` and
+  ``bsoncxx::stdx::optional<T>`` as aliases for Boost or MNMLSTC/core
+  library equivalents. For context, see the :ref:`Build System Breaking Changes
+  <version-4.0-build-system-breaking>` section.
+
+ABI Breaking Changes in v4.0
+````````````````````````````
+
+The v4.0 driver introduces the following breaking changes to the ABI:
+
+- Drops support for configuring ``bsoncxx::stdx::string_view`` and
+  ``bsoncxx::stdx::optional<T>`` as aliases for Boost or MNMLSTC/core
+  library equivalents. For context, see the :ref:`Build System Breaking Changes
+  <version-4.0-build-system-breaking>` section.
+
 .. _version-3.11-breaking-changes:
 
-Version 3.11 Breaking Changes
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Version 3.11
+~~~~~~~~~~~~
 
-This driver version introduces the following breaking changes:
+Build System Breaking Changes in v3.11
+``````````````````````````````````````
+
+The v3.11 driver introduces the following breaking changes to the build system:
 
 - Drops support for {+mdb-server+} v3.6.
 - Requires MongoDB C Driver v1.28.0 or later.
+
+ABI Breaking Changes in v3.11
+`````````````````````````````
+
+The v3.11 driver introduces the following breaking changes to the ABI:
+
 - Removes export of the following private member functions in the bsoncxx
   ABI:
-   
-  - ``bsoncxx::v_noabi::types::bson_value::value::value(const uint8_t*, uint32_t, uint32_t, uint32_t)``
+
+  - ``bsoncxx::v_noabi::types::bson_value::value::value(const uint8_t*,
+    uint32_t, uint32_t, uint32_t)``
```

Comment:
FYI line-breaks in source code do not appear to translate to line-breaks in the rendered page (the ToC overlap issue is not resolved). I am fine with this being deferred to a separate ticket/PR however.

Reply:
Made a ticket: https://jira.mongodb.org/browse/DOCSP-45596